common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -1073,6 +1070,9 @@ public void onDeviceUpdate(TransportProtos.SessionInfoProto sessionInfo, Device
     @Override
     public void onDeviceDeleted(DeviceId deviceId) {
         doDisconnect();
```

Comment:
I believe, the doDisconnect is redundant here. The ctx.close will invoke "doDisconnect", because it will trigger "operationComplete". Please check this and improve the code.

Reply:
You are right, I will remove this call.